tmk_core/protocol/vusb/vusb.c

```diff
@@ -509,7 +516,37 @@ const PROGMEM uchar shared_hid_report[] = {
     0x95, 0x01,                //   Report Count (1)
     0x75, 0x10,                //   Report Size (16)
     0x81, 0x00,                //   Input (Data, Array, Absolute)
-    0xC0                       // End Collection
+    0xC0,                      // End Collection
+
+    // padded to 16 bits to work with the existing machinery
+    0x09, 0x01,                          // Usage (Consumer Control)
+    0xA1, 0x01,                          // Collection (Application)
+    0x85, REPORT_ID_PROGRAMMABLE_BUTTON, //   Report ID
+    0x09, 0x03,                          //   Usage (Programmable Buttons)
+    0xA1, 0x04,                          //   Collection (Named Array)
+    0x05, 0x09,                          //     Usage Page (Button)
+    0x09, 0x01,                          //     Button 1
+    0x09, 0x02,                          //     Button 2
+    0x09, 0x03,                          //     Button 3
+    0x09, 0x04,                          //     Button 4
+    0x09, 0x05,                          //     Button 5
+    0x09, 0x06,                          //     Button 6
+    0x09, 0x07,                          //     Button 7
+    0x09, 0x08,                          //     Button 8
+    0x09, 0x09,                          //     Button 9
+    0x09, 0x0a,                          //     Button 10
+    0x09, 0x0b,                          //     Button 11
+    0x09, 0x0c,                          //     Button 12
+    0x09, 0x0d,                          //     Button 13
+    0x09, 0x0e,                          //     Button 14
+    0x09, 0x0f,                          //     Button 15
```

Comment:
> And here you declare support for 16 programmable buttons, even though you reserved just 3 keycodes.

Indeed, this was meant for demonstration purposes. As soon as the details are decided on, I'll unify this.

> Is it possible to use “Usage Minimum” and “Usage Maximum” here, like for the system control and consumer reports, instead of listing every usage code?

I'll test that.